dapp/src/components/AccountStatusDropdown.js

```diff
@@ -33,15 +33,17 @@ const AccountStatusDropdown = ({ className, showLogin, dapp }) => {
           className={`account-status d-flex justify-content-center align-items-center clickable ${className} ${
             open ? 'open' : ''
           }`}
-          onClick={(e) => {
+          onClick={async (e) => {
             e.preventDefault()
             if (dapp && !active) {
               showLogin()
             } else if (active && !correctNetwork) {
               analytics.track('On Change network', {
                 category: 'settings',
               })
-              switchEthereumChain()
+              // open the dropdown to allow disconnecting, while also requesting an auto switch to mainnet
+              await switchEthereumChain()
```

Comment:
nit: should this be wrapped inside a try catch block?

Reply:
This in effect [calls this function](https://github.com/OriginProtocol/origin-dollar/blob/master/dapp/src/utils/web3.js#L18) on the window object and from the documentation it does not throw an error. Rather returns null on success and returns error on fail: https://eips.ethereum.org/EIPS/eip-3326

Anyway your comment made me check, because I wasn't sure before :)